src/components/MovieAdder.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, ChangeEvent } from "react";
+import AsyncSelect from "react-select/lib/Async";
+import { debounce } from "underscore";
+
+import { MoviePreference, User } from "../types";
+
+const promiseOptions = (inputValue: string) => {
+  const searchParams = new URLSearchParams({ search: inputValue });
+  return fetch("/search-movies?" + searchParams.toString())
+    .then(response => response.json())
+    .then(function(json: {
+      searchResults: Array<{ id: string; title: string }>;
+    }) {
+      return json.searchResults.map(option => ({
+        label: option.title,
+        value: option.id
+      }));
+    });
+};
+
+interface externalMovie {
+  value: number;
+  label: string;
+}
+
+const MovieAdder = (props: { user: User }): JSX.Element => {
+  const [externalMovie, setExternalMovie] = useState<externalMovie | null>(
+    null
+  );
+  const handleChange = (value: any, data: { action: string }) => {
+    if (data.action === "select-option") {
+      setExternalMovie(value as externalMovie);
```

Comment:
If we know that value is going to always be an externalMovie, then we could do `const handleChange = (value: externalMovie, data: ...`.

Reply:
What's best practice here? Technically there are a bunch of change actions I think, and we only want to listen to the one

Reply:
And the others would have different shapes

Reply:
Ahhhh ok. I think it like really anal land, we would have types for each of the possible values we could get back, and another type that is the set of all those types, but it's a PITA and I think this is fine for now